lib/msal-common/src/auth/IdToken.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ClientAuthError } from "../error/ClientAuthError";
+import { StringDict } from "../utils/MsalTypes";
+import { StringUtils } from "../utils/StringUtils";
+import { ICrypto } from "../utils/crypto/ICrypto";
+
+/**
+ * @hidden
+ */
+export class IdToken {
+
+    issuer: string;
+    objectId: string;
+    subject: string;
+    tenantId: string;
+    version: string;
+    preferredName: string;
+    name: string;
+    homeObjectId: string;
+    nonce: string;
+    expiration: string;
+    rawIdToken: string;
+    claims: StringDict;
+    sid: string;
+    cloudInstance: string;
+    /* tslint:disable:no-string-literal */
+    constructor(rawIdToken: string, crypto: ICrypto) {
+        if (StringUtils.isEmpty(rawIdToken)) {
+            throw ClientAuthError.createIdTokenNullOrEmptyError(rawIdToken);
+        }
+        try {
+            this.rawIdToken = rawIdToken;
+            this.claims = IdToken.extractIdToken(rawIdToken, crypto);
+            if (this.claims) {
+                if (this.claims.hasOwnProperty("iss")) {
```

Comment:
We might want to consider having enumerations/constants for these strings, even if they aren't being used elsewhere. Disabling the linter like this feels pretty bad if that's the configuration we've set up.

Reply:
+1, I will make some changes.